client/src/components/TreeTable/TreeTable.js

```diff
@@ -40,148 +32,122 @@ const styles = theme => ({
 
 class TreeTable extends Component {
-
   constructor(props) {
-    super(props)
+    super(props);
     this.state = {
       detailsPane: false,
-    }
-    this.onChangeRowsPerPage = this.onChangeRowsPerPage.bind(this)
+      page: 0
+    };
   }
 
   componentDidMount() {
     const payload = {
-      page: this.props.page,
+      page: this.state.page,
       rowsPerPage: this.props.rowsPerPage,
       order: this.props.order,
       orderBy: this.props.orderBy
-    }
-    this.props.getTreesAsync(payload)
-  }
-
-  handleSelection = (e) => {
-
+    };
+    this.props.getTreesAsync(payload);
   }
 
-  handleClick = (event, id) => {
-    const { selected } = this.state
-    const selectedIndex = selected.indexOf(id)
-    let newSelected = []
-    if (selectedIndex === -1) {
-      newSelected = newSelected.concat(selected, id)
-    } else if (selectedIndex === 0) {
-      newSelected = newSelected.concat(selected.slice(1))
-    } else if (selectedIndex === selected.length - 1) {
-      newSelected = newSelected.concat(selected.slice(0, -1))
-    } else if (selectedIndex > 0) {
-      newSelected = newSelected.concat(
-        selected.slice(0, selectedIndex),
-        selected.slice(selectedIndex + 1),
-      )
-    }
-    this.setState({ selected: newSelected })
-  }
-
-  toggleDrawer = (event, id) => {
-    event.preventDefault();
-    if (id !== undefined) this.props.getTreeAsync(id);
+  toggleDrawer = id => {
+    console.log(id);
+    this.props.getTreeAsync(id);
     const { detailsPane } = this.state;
     this.setState({
       detailsPane: !detailsPane
     });
-  }
+  };
 
-  onPageChange = (event, page) => {
-    this.props.getTreesAsync({ page: page, rowsPerPage: this.props.rowsPerPage })
-  }
+  onPageChange = (page, rowsPerPage) => {
+    this.props.getTreesAsync({
+      page,
+      rowsPerPage
+    });
+    this.setState({
+      page
+    });
+  };
 
-  onChangeRowsPerPage = event => {
-    this.props.getTreesAsync({ page: this.props.page, rowsPerPage: event.target.value })
-  }
+  onChangeRowsPerPage = rowsPerPage => {
+    this.props.getTreesAsync({
+      page: this.state.page,
+      rowsPerPage
+    });
+  };
 
-  isSelected = (id) => {
-    return this.props.selected.indexOf(id) !== -1
-  }
+  onSort = (column, ascending) => {
+    this.props.sortTrees(column, ascending);
+  };
 
   render() {
-    const { numSelected, classes, rowsPerPage, selected, order, orderBy, treesArray, getLocationName, treeCount, byId, tree } = this.props
+    const { treesArray, tree } = this.props;
+    const columnData = [
+      { name: 'id', label: 'Id', options: { sortable: true } },
+      { name: 'timeCreated', label: 'Creation', options: { sortable: true } },
+      { name: 'timeUpdated', label: 'Updated', options: { sortable: true } },
+      {
+        label: 'Location',
+        options: {
+          sortable: false,
+          customBodyRender: () => {
+            return <span>botched rendering func</span>; // TODO: replace
```

Comment:
is this todo for this PR or a future one?

Reply:
Future, location rendering is not implemented on master to my knowledge